kartothek/io/testing/index.py

```diff
@@ -40,7 +33,7 @@ def test_build_indices(store_factory, metadata_version, bound_build_dataset_indi
     # Assert indices are properly created
     dataset_factory = DatasetFactory(dataset_uuid, store_factory, load_all_indices=True)
     expected = {2: ["cluster_1", "cluster_2"], 3: ["cluster_2"], 1: ["cluster_1"]}
-    assert_index_dct_equal(expected, dataset_factory.indices["p"].index_dct)
+    assert expected == dataset_factory.indices["p"].index_dct
```

Comment:
@crepererum What I meant to say is that `dataset_factory.indices["p"].index_dct` and `expected` here may have a different order for keys as well as values (the list of partition labels is not guaranteed to be ordered)

Reply:
The dict comparison does indeed not care about order of keys, but the values will be interpreted as different because we are comparing e.g. `["cluster_1", "cluster_2"] == ["cluster_2", "cluster_1"]`

Reply:
ah 👍 
But still, if @fjetter wants to extend the usage of toolz, there is no point in having a "remove it from tests" PR